norddrop/src/device/mod.rs

```diff
@@ -593,36 +595,71 @@ fn open_database(
     dbpath: &str,
     events: &EventDispatcher,
     logger: &slog::Logger,
+    moose: &Arc<dyn drop_analytics::Moose>,
 ) -> Result<drop_storage::Storage> {
     // Try opening the DB 3 times
     for i in 1..=3 {
         match drop_storage::Storage::new(logger.clone(), dbpath) {
             Ok(storage) => return Ok(storage),
-            Err(err) => warn!(logger, "Failed to open DB: {err}, already tried {i} times",),
+            Err(err) => error!(
+                logger,
+                "Failed to open DB at \"{dbpath}\": {err}, already tried {i} times",
+            ),
         }
```

Comment:
I considered adding a moose call here as well, but i don't think it would give enough value given how much spam it would introduce

Reply:
Not worth logging to moose if we iterate(its probably gonna be the same error multiple times) imho

Reply:
Thinking about it twice - it would be nice to know at least once that this happened as other logs might not be exercised if everything went fine. This way we're left oblivious to the fact that the database failed but it was properly flushed and recreated.

Thinking of it - we can emit a moose event here but remove the retry. It could be like this:
- try to use a database
- if fail -> emit moose event(1), try to recreate it 
- if fail recreating -> emit moose event(2) and use `:memory:`
- if fail using `:memory:` -> emit moose event(3) and fail with an error

Reply:
done

Reply:
@Lipt0nas what do you think about my following(after the initial comment) suggestion?

Reply:
I left the retrying part in, but otherwise it's essentially what you described, we try to open, if it doesn't work, we try to use `:memory:` and inform of that fact in the description of the event, and if that fails, we emit and event and error out

Reply:
We still have `for i in 1..=3 {...}` present, which I feel is unecessary

Reply:
removed it